build.sh

```diff
@@ -97,65 +97,82 @@ cd EmulatorJS
-compileProject "gearcoleco" "https://github.com/EmulatorJS/Gearcoleco.git" "./platforms/libretro/" "Makefile"
-compileProject "parallel-n64" "https://github.com/EmulatorJS/parallel-n64.git" "./" "Makefile"
-compileProject "mame2003-plus" "https://github.com/EmulatorJS/mame2003-plus-libretro.git" "./" "Makefile"
-compileProject "puae" "https://github.com/EmulatorJS/libretro-uae.git" "./" "Makefile"
-compileProject "vice_x64" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=x64"
-compileProject "vice_x64sc" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=x64sc"
-compileProject "vice_x128" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=x128"
-compileProject "vice_xpet" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=xpet"
-compileProject "vice_xplus4" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=xplus4"
-compileProject "vice_xvic" "https://github.com/EmulatorJS/vice-libretro.git" "./" "Makefile" "EMUTYPE=xvic"
-compileProject "smsplus-gx" "https://github.com/EmulatorJS/smsplus-gx" "./" "Makefile.libretro"
-
-cd "RetroArch"
-git pull
-cd "dist-scripts"
+compileStartPath="$PWD"
+for row in $(jq -r '.[] | @base64' ../cores.json); do
+    cd $compileStartPath
+
+    _jq() {
+        echo ${row} | base64 --decode | jq -r ${1}
+    }
+
+    name=`echo $(_jq '.') | jq -r '.name'`
+    repo=`echo $(_jq '.') | jq -r '.repo'`
+    license=`echo $(_jq '.') | jq -r '.license'`
+    buildpath=`echo $(_jq '.') | jq -r '.makeoptions.buildpath'`
+    makescript=`echo $(_jq '.') | jq -r '.makeoptions.makescript'`
+    arguments=`echo $(_jq '.') | jq -r '.makeoptions.arguments[] | @base64'`
+
+    argumentstring=""
+    for rowarg in $(echo "${arguments}"); do
+        argumentstring="$argumentstring `echo $rowarg | base64 --decode`"
+    done
+
+    echo "Starting compile of core $name"
+    echo "Working dir $PWD"
+
+    compileProject "$name" "$repo.git" "$buildpath" "$makescript" "$argumentstring"
+
+    # write JSON stanza for this core to disk
+    echo ${row} | base64 --decode > "./core.json"
+
+    if [ ! -z "$license" -a "$license" != " " ]; then
+        # license file is provided - copy it
+        echo "License file: $name/$license"
+        cp $name/$license "./license.txt"
+    fi
 
-mv core-temp/normal/*.bc ./
-emmake ./build-emulatorjs.sh emscripten clean no no
-rm -f *.bc
+    echo "Build wasm's for core $name"
+    cd "RetroArch"
+    git pull
+    cd "dist-scripts"
 
-mv core-temp/threads/*.bc ./
-emmake ./build-emulatorjs.sh emscripten clean yes no
-rm -f *.bc
+    mv core-temp/normal/*.bc ./
+    emmake ./build-emulatorjs.sh emscripten clean no no
+    rm -f *.bc
 
-mv core-temp/legacy/*.bc ./
-emmake ./build-emulatorjs.sh emscripten clean no yes
-rm -f *.bc
+    mv core-temp/threads/*.bc ./
+    emmake ./build-emulatorjs.sh emscripten clean yes no
```

Comment:
Could we dump the output from these emmake calls to a log file instead of to the console?

```bash
rm build.sh
emmake ./build-emulatorjs.sh emscripten clean yes no >> build.sh
```

Reply:
Compile and emake logs are now sent to a log file named for the core being built.